src/domain/Loan.ts

```diff
@@ -62,7 +63,15 @@ class Installments {
     this.createdAt = createdAt;
     this.updatedAt = updatedAt;
     this.status = status;
+    this.paidAt = this.paidAt;
   }
+
+  paidInstallment = () => {
+    if (this.status === "paid") {
```

Comment:
@daniszcode alguns pontos aqui:
* a função **paidInstallment** deve ser **void**, sendo assim ela não possui retorno (semelhante ao **createInstallmentPlan**)
* Nela você deve apenas atualizar os valores do **status** e **paidAt** mesmo.
* Se o status já está como **Paid** você não precisa fazer nada. Só deve atualizar os campos se o status estiver diferente de **Paid**

Reply:
O status só fica paid quando a gente instancia o objeto e coloca como paid, não é isso? Então a minha logica foi pra ele setar como paid, só quando instanciado dessa forma 🤔

Reply:
@daniszcode , é que se o status já está paid, quer dizer que ele foi anteriormente pago. Então nao precisaria mudar o valor novamente para **paid** e nem colocar a data de novo.
Essa função vai ser chamada quando você quiser pagar uma parcela. Então quando ela for chamada a primeira vez, o status estará como **open**. Daí o if precisa verificar se o status é **diferente de paid**, para aí sim ele atualizar o status para **paid** e colocar a data no paidAt.
Caso contrario, se o status já estiver paid, você nao precisa mais mudar nada pq já está tudo certinho :)

E essa função sendo **void** não deve ter **return** ;)

Reply:
Eu tinha ignorado o fato que ele antes de ir para 'paid', se manteria 'open'. Mas agora acho que entendi! = )
Obrigada, jedi!